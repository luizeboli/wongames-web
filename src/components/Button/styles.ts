import { darken } from 'polished';
import styled, { css, DefaultTheme } from 'styled-components';

import { ButtonProps } from '.';

type ContainerProps = {
  hasIcon: boolean;
} & Pick<ButtonProps, 'size' | 'fullWidth' | 'minimal'>;

const containerModifiers = {
  small: (theme: DefaultTheme) => css`
    height: 3rem;
    font-size: ${theme.font.sizes.xsmall};
    padding: ${theme.spacings.xxsmall};
  `,
  medium: (theme: DefaultTheme) => css`
    height: 4rem;
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacings.xxsmall} ${theme.spacings.medium};
  `,
  large: (theme: DefaultTheme) => css`
    height: 5rem;
    font-size: ${theme.font.sizes.medium};
    padding: ${theme.spacings.xxsmall} ${theme.spacings.xlarge};
  `,
  fullWidth: () => css`
    width: 100%;
  `,
  withIcon: (theme: DefaultTheme) => css`
    svg {
      width: 1.5rem;
      height: 1.5rem;

      & + span {
        margin-left: ${theme.spacings.xxsmall};
      }
    }
  `,
  minimal: (theme: DefaultTheme) => css`
    background: none;
    color: ${theme.colors.primary};

    &:hover {
      background: none;
      color: ${darken(0.1, theme.colors.primary)};
    }
  `,
};

export const Container = styled.button<ContainerProps>`
  ${({ theme, size, fullWidth, hasIcon, minimal }) => css`
    cursor: pointer;
    display: inline-flex;
    align-items: center;
    justify-content: center;
    color: ${theme.colors.white};
    background: linear-gradient(180deg, #ff5f5f 0%, #f062c0 50%);
    border: none;
    border-radius: ${theme.border.radius};
    text-decoration: none;

    :hover {
      background: linear-gradient(180deg, #e35565 0%, #d958a6 50%);
    }

    ${!!size && containerModifiers[size](theme)}
    ${fullWidth && containerModifiers.fullWidth()}
    ${hasIcon && containerModifiers.withIcon(theme)}
    ${minimal && containerModifiers.minimal(theme)}
  `}
`;
