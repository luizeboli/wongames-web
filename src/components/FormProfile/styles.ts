import styled from 'styled-components';

export const Container = styled.main``;

export const Form = styled.form``;
