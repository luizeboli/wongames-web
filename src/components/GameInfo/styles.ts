import styled from 'styled-components';

export const Container = styled.main``;

export const Description = styled.p``;

export const ButtonsWrapper = styled.div``;
