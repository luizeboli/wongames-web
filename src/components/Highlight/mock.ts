export default {
  title: 'Read Dead is back',
  subtitle: 'Come buy this funny game',
  buttonLabel: 'Buy!',
  buttonLink: '/red-dead',
  backgroundImage: '/img/red-dead-img.jpg',
};
