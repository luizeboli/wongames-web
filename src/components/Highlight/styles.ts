import styled, { css } from 'styled-components';
import media from 'styled-media-query';

import { HighlightProps } from '.';

type WrapperProps = Pick<HighlightProps, 'backgroundImage' | 'alignment'>;

const wrapperModifiers = {
  right: () => css`
    grid-template-areas: 'floatimage container';
    grid-template-columns: 1.3fr 2fr;

    ${Container} {
      text-align: right;
    }
  `,
  left: () => css`
    grid-template-areas: 'container floatimage';
    grid-template-columns: 2fr 1.3fr;

    ${Container} {
      text-align: left;
    }

    ${FloatImage} {
      justify-self: flex-end;
    }
  `,
};

export const Wrapper = styled.section<WrapperProps>`
  ${({ backgroundImage, alignment }) => css`
    position: relative;
    height: 23rem;
    background-image: url(${backgroundImage});
    background-position: center;
    background-size: cover;
    display: grid;

    &::after {
      content: '';
      position: absolute;
      width: 100%;
      height: 100%;
      background-color: rgba(0, 0, 0, 0.6);
    }

    ${wrapperModifiers[alignment!]()}

    ${media.greaterThan('medium')`
      height: 32rem;
    `}
  `}
`;

export const Container = styled.div`
  ${({ theme }) => css`
    grid-area: container;
    z-index: ${theme.layers.base};
    padding: ${theme.spacings.xsmall};

    ${media.greaterThan('medium')`
      align-self: end;
      padding: ${theme.spacings.large};
    `}
  `}
`;

export const Title = styled.h2`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.large};
    font-weight: ${theme.font.bold};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.xxlarge};
    `}
  `}
`;

export const Subtitle = styled.h3`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.small};
    font-weight: ${theme.font.light};
    margin-bottom: ${theme.spacings.medium};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.large};
    `}
  `}
`;

export const FloatImage = styled.img`
  ${({ theme }) => css`
    grid-area: floatimage;
    align-self: flex-end;
    z-index: ${theme.layers.base};

    max-height: 23rem;
    max-width: 100%;

    ${media.greaterThan('medium')`
      max-height: 32rem;
    `}
  `}
`;
