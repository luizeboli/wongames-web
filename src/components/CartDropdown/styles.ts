import styled from 'styled-components';

import { Container as CartContainerStyles } from 'components/CartList/styles';

export const Wrapper = styled.main`
  ${CartContainerStyles} {
    width: 56rem;
  }
`;
