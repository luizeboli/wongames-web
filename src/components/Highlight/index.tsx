import Image from 'next/image';

import Button from 'components/Button';

import * as S from './styles';

export type HighlightProps = {
  title: string;
  subtitle: string;
  buttonLabel: string;
  buttonLink: string;
  backgroundImage: string;
  floatImage?: string;
  alignment?: 'right' | 'left';
};

const Highlight = ({ title, subtitle, buttonLabel, buttonLink, backgroundImage, floatImage, alignment = 'right' }: HighlightProps) => (
  <S.Wrapper alignment={alignment} data-cy="highlight">
    <Image src={backgroundImage} alt={title} layout="fill" objectFit="cover" />
    {!!floatImage && (
      <S.FloatImageWrapper>
        <Image src={floatImage} alt={title} width={400} height={300} objectFit="contain" />
      </S.FloatImageWrapper>
    )}
    <S.Container>
      <S.Title>{title}</S.Title>
      <S.Subtitle>{subtitle}</S.Subtitle>
      <Button as="a" href={buttonLink}>
        {buttonLabel}
      </Button>
    </S.Container>
  </S.Wrapper>
);

export default Highlight;
