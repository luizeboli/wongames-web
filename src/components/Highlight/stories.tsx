import { Meta, Story } from '@storybook/react/types-6-0';

import item from './mock';

import Highlight, { HighlightProps } from '.';

export default {
  title: 'Highlight',
  component: Highlight,
  args: {
    ...item,
  },
} as Meta;

export const Basic: Story<HighlightProps> = (args) => <Highlight {...args} />;

export const withFloatImage: Story<HighlightProps> = (args) => <Highlight {...args} />;

withFloatImage.args = {
  floatImage: '/img/red-dead-float.png',
};
