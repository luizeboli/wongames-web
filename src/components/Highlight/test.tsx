import { render, screen } from 'utils/test-utils';

import * as S from './styles';

import Highlight from '.';

const props = {
  title: 'Heading 1',
  subtitle: 'Heading 2',
  buttonLabel: 'Buy Now',
  buttonLink: '/buy-now',
  backgroundImage: '/img/red-dead-img.jpg',
};

describe('<Highlight />', () => {
  it('should render correctly', () => {
    const { container } = render(<Highlight {...props} />);

    expect(screen.getByRole('heading', { name: /Heading 1/i })).toBeInTheDocument();

    expect(screen.getByRole('heading', { name: /Heading 2/i })).toBeInTheDocument();

    expect(screen.getByRole('link', { name: /Buy Now/i })).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should render background image', () => {
    render(<Highlight {...props} />);

    expect(screen.getByRole('img')).toHaveAttribute('src', '/img/red-dead-img.jpg');
  });

  it('should render float image', () => {
    render(<Highlight {...props} floatImage="/float-image.png" />);

    expect(screen.getAllByRole('img', { name: props.title })[1]).toHaveAttribute('src', '/float-image.png');
  });

  it('should render a float image aligned on left by default', () => {
    const { container } = render(<Highlight {...props} floatImage="/float-image.png" />);

    expect(container.firstChild).toHaveStyleRule('grid-template-areas', "'floatimage container'");

    expect(container.firstChild).toHaveStyleRule('text-align', 'right', {
      modifier: `${S.Container}`,
    });
  });

  it('should render a float image aligned on right', () => {
    const { container } = render(<Highlight {...props} floatImage="/float-image.png" alignment="left" />);

    expect(container.firstChild).toHaveStyleRule('grid-template-areas', "'container floatimage'");

    expect(container.firstChild).toHaveStyleRule('text-align', 'left', {
      modifier: `${S.Container}`,
    });
  });
});
