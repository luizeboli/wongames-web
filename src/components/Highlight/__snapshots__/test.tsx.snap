// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Highlight /> should render correctly 1`] = `
.c0 {
  position: relative;
  height: 23rem;
  display: grid;
  grid-template-areas: 'floatimage container';
  grid-template-columns: 1.3fr 2fr;
}

.c0 img {
  position: absolute;
}

.c0::after {
  content: '';
  position: absolute;
  width: 100%;
  height: 100%;
  background-color: rgba(0,0,0,0.6);
}

.c0 .c1 {
  text-align: right;
}

.c2 {
  grid-area: container;
  z-index: 10;
  padding: 1.6rem;
}

.c3 {
  color: #FAFAFA;
  font-size: 1.8rem;
  font-weight: 600;
}

.c4 {
  color: #FAFAFA;
  font-size: 1.4rem;
  font-weight: 300;
  margin-bottom: 3.2rem;
}

.c5 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #FAFAFA;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  border: none;
  border-radius: 0.4rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c5:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

@media (min-width:768px) {
  .c0 {
    height: 32rem;
  }
}

@media (min-width:768px) {
  .c2 {
    -webkit-align-self: end;
    -ms-flex-item-align: end;
    align-self: end;
    padding: 4.0rem;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 1.8rem;
  }
}

<div>
  <section
    class="c0"
  >
    <img
      alt="Heading 1"
      layout="fill"
      src="/img/red-dead-img.jpg"
    />
    <div
      class="c1 c2"
    >
      <h2
        class="c3"
      >
        Heading 1
      </h2>
      <h3
        class="c4"
      >
        Heading 2
      </h3>
      <a
        class="c5"
        href="/buy-now"
      >
        <span>
          Buy Now
        </span>
      </a>
    </div>
  </section>
</div>
`;
